server/implementation/src/main/java/io/smallrye/graphql/bootstrap/Bootstrap.java

```diff
@@ -133,6 +135,7 @@ public static GraphQLSchema bootstrap(Schema schema, boolean skipInjectionValida
 
     private Bootstrap(Schema schema, boolean skipInjectionValidation) {
         this.schema = schema;
+        this.linkProcessor = new LinkProcessor(schema);
```

Comment:
I'd say this should be wrapped into 
`if (Config.get().isFederationEnabled()) {`
and stay null if federation is disabled, no?

Reply:
And then all usages of the `linkProcessor` should similary be wrapped into a condition.

Reply:
Hmm, I see you have `Config.get().isFederationEnabled()` inside the `LinkProcessor` class but it would still be better to avoid using that class altogether, for optimization purposes (it can be eliminated as dead code by GraalVM)

Reply:
In `Bootstrap`, I also use `linkProcessor.newNameDirective` and `linkProcessor.newName`. If `LinkProcessor` is not defined, this operation will fail, so they should probably be checked as well.

Reply:
Yes, I would wrap usage of `linkProcessor` into an `if`. And only call it when Federation is enabled.
The `LinkProcessor` itself won't have to check whether Federation is enabled anymore, it can assume that it is enabled.

Reply:
This should be fixed now.